hedera-node/hedera-mono-service/src/main/java/com/hedera/services/ledger/TransferLogic.java

```diff
@@ -150,19 +150,23 @@ public void doZeroSum(final List<BalanceChange> changes) {
             }
         }
 
-        if (validity == OK
-                && (autoCreationFee > 0)
-                && (autoCreationFee > (long) accountsLedger.get(txnCtx.activePayer(), BALANCE))) {
-            validity = INSUFFICIENT_PAYER_BALANCE;
-        }
-
         if (validity == OK) {
             adjustBalancesAndAllowances(changes);
             if (autoCreationFee > 0) {
-                payAutoCreationFee(autoCreationFee);
-                autoCreationLogic.submitRecordsTo(recordsHistorian);
+                // try to pay auto creation fee only **after** balances have been adjusted,
```

Comment:
Amazing catch! 👏 ❤️ 

But do note the rollback below,
```
    accountsLedger.undoChangesOfType(List.of(BALANCE));
```
...will leave the allowance changes still committed. 

Maybe we should just track if any of the `BalanceChange`'s are updating the payer's balance? (And if so, compare the `autoCreationFee` to that updated balance instead of `accountsLedger.get(BALANCE)`?)